pkg/types/config/validate.go

```diff
@@ -324,13 +324,9 @@ func ValidateDomainName(v string) error {
 	}
 
 	split := strings.Split(v, ".")
-	for i, segment := range split {
-		// Trailing dot is OK
```

Comment:
note to self: Comments are usually a hint that the initial dev felt this was surprising/important.  But checking the Git logs, this is from d9359cb0 (coreos/tectonic-installer#3085), and neither the commit message nor PR discussion has anything to say about why they specifically allowed trailing dots then.